### ISSUE TITLE
Information about interaction target in SecondariesVector missing

### ISSUE THREAD

**Issue by Jean1995** [tudo-astroparticlephysics/PROPOSAL#221]
If we want to calculate a list of `ParticleState` from a `StochasticLoss` object after propagation, for example in the case of [MuonPairProduction](https://github.com/tudo-astroparticlephysics/PROPOSAL/blob/3bda79a4e05280b09def6066f9594d27923ede29/src/PROPOSAL/PROPOSAL/secondaries/parametrization/mupairproduction/KelnerKokoulinPetrukhinMupairProduction.h#L33), we need the information about the target (Component) that we interacted with. However, this information is missing in the output of our Propagator (`Secondaries`), so it is not possible to use this function.

We need to add this information for each StochasticInteraction.

**Comment by Jean1995** [comment]
In what form should we include the information about the target that we interacted with?

My first idea would be to include the hash of the target. If we interacted with a Component, we include the hash of the Component (where we can trace back the Component object using the component_map). If we interacted with a Medium (i.e. if we have Ionization), we could include the hash of the medium. However, there is no way at the moment to trace back the original medium yet.
